β-normalize all inferred types (#745)
dhall-lang/dhall-lang: .github/CONTRIBUTING.md

- If an expression type-checks, normalizing that expression won't diverge
- Normalizing an inferred type won't diverge
- Normalizing an expression doesn't change its type

> - An inferred type is in β-normal form

Author: Would this be clearer?

```suggestion
- All inferred types are in β-normal form
```

Reviewer: Yeah
